client/serviceregistration/address.go

```diff
@@ -8,22 +8,49 @@ import (
 	"github.com/hashicorp/nomad/plugins/drivers"
 )
 
-// GetAddress returns the IP and port to use for a service or check. If no port
-// label is specified (an empty value), zero values are returned because no
-// address could be resolved.
+// GetAddress returns the IP (or custom advertise address) and port to use for a
+// service or check registration. If no port label is specified (an empty value),
+// zero values are returned because no address could be resolved.
 func GetAddress(
```

Comment:
Nothing for this PR, but noting for the future that I wonder if it's worth changing this function signature to take a struct rather than a growing list of types.